src/Microsoft.AspNet.Http.Interfaces/IRequestIdentifierFeature.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Http.Interfaces
+{
+    /// <summary>
+    /// Feature to identify a request.
+    /// </summary>
+    public interface IRequestIdentifierFeature
+    {
+        /// <summary>
+        /// Identifier to trace a request.
+        /// </summary>
+        Guid TraceIdentifier { get; }
```

Comment:
GUIDs actually solve the problem nicely because they have a definite ordering so can be indexed. You can't do that with strings without a lot of ceremony: making sure everything is length-prefixed in storage (nvarchar, but then you kill indexability), making sure everybody agrees on which comparer to use (good luck doing this in SQL), and so on.